### ISSUE TITLE
RARBG bencoding with qbittorrent

### ISSUE THREAD

**Issue by mcaps1** [Jackett/Jackett#4584]
Hello,

I am currently having an issue using the blackhole directory with RARBG torrents. When I select a torrent to send to the blackhole folder, qbittorrent rejects the file. When I add torrents from other trackers, it works perfectly fine. When I try to open a .torrent file manually that Jackett placed in the Blackhole folder, I get a bencoding error. Downloading torrent files directly from RARBG are working perfectly fine for me though. I have no other issues with Jackett and magnet links work well.

Jackett v 0.10.692.0
Windows Server 2016
Caddy Webserver (but issue happens on 127.0.0.1 also)

**Comment by minitoine** [comment]
Have the same issue with Sonarr. Grabbing via Sonarr gives an error, doing it manually within Jackett is fine.

**Comment by ukharley** [comment]
I also can't get recent stuff from RARBG. My error is torrent file is empty. I had a similar issue a few months back, due to Jackett not scraping RARBG but instead using https://torrentapi.org/. Seems it's doing so again.

**Comment by ukharley** [comment]
I see RARBG has sorted this in a closed issue.

**Comment by garfield69** [comment]
As you may know, Jackett's RarBG indexer uses the RarBG API to fetch results, since it's prohibited from using the RarBG WEB site via web page scraping methods.
When the RarBG API returns results for a torrent search, it provides a direct download link from the WEB RarBG site for each torrent found.
So when you use a download link you in effect attempt to access the RarBG WEB site at this point.
However, for your IP block in your region, RarBG WEB is invoking cloudflare DDOS protection checking.
The cloudflare DDOS page is interfering with your blackhole directory dump, and the other users Sonarr grab.
Instead of getting a .torrent data file, you are getting the cloundflare DDOS HTML page, and hence the bencoding error.
Don't know if there is anything we can do about this from the Jackett side of the process, 
@kaso17 any ideas?

**Comment by minitoine** [comment]
Is the DDos protection from RarBG a new feature ? It worked like a charm for the last few months..

**Comment by VirtuaWin** [comment]
The same error that @minitoine described is happening to me on qbittorrent.
Qbittorrent can get the RSS feeds just fine from Jackett, but when downloading the torrents (as @garfield69 said) qbittorrent has to access RARBG’s WEB and that's when I get the error.

When I first noticed this behavior I searched for the torrents directly in Jackett's WEBUI, and one thing that I noticed is that when trying to download the torrent file it's indeed is not possible (since it is accessing RARBG's WEB), but on the other hand the magnet link works just fine (if I copy it into qbittorrent the download starts right away).

With that being said, it seems to me that a solution to this problem would be to configure Jackett to return magnet links for the RARBG RSS feed, instead of links to RARBG's WEB .torrent file. Is that currently possible?

**Comment by garfield69** [comment]
yes its possible. we could add to the rarbg indexer config an option to make the magnet the default download-link instead of the .torrent.
but we will have to wait for @kaso17  or another c# coder to implement, as its beyond my limited c# skills.

**Comment by VirtuaWin** [comment]
> yes its possible. we could add to the rarbg indexer config an option to make the magnet the default download-link instead of the .torrent.
> but we will have to wait for @kaso17 or another c# coder to implement, as its beyond my limited c# skills.

That would be awesome.

**Comment by minitoine** [comment]
Here are logs from the click on blackhole download to the succesful download of the torrent file. But the .torrent file is considered as invalid.


```
Feb 15, 2019 03:58:28 PM | Debug | 
WebClient(HttpWebClient): Returning OK => 3311 bytes

Feb 15, 2019 03:58:26 PM | Debug | 
WebClient(HttpWebClient): delaying request for https://rarbg.to/threat_defence.php?defence=1&r=40540307 by 2.099086 seconds

Feb 15, 2019 03:58:26 PM | Debug | 
WebClient(HttpWebClient).GetBytes(Url:https://rarbg.to/threat_defence.php?defence=1&r=40540307)

Feb 15, 2019 03:58:26 PM | Debug | 
WebClient(HttpWebClient): Returning Redirect => https://rarbg.to/threat_defence.php?defence=1&r=40540307 0 bytes

Feb 15, 2019 03:58:26 PM | Debug | 
[MONO relative redirect bug] Rewriting relative redirect URL from file:///threat_defence.php%3Fdefence=1&r=40540307 to https://rarbg.to/threat_defence.php?defence=1&r=40540307

Feb 15, 2019 03:58:23 PM | Debug | 
WebClient(HttpWebClient): delaying request for https://rarbg.to/download.php?id=5gs7zj9&f=jackett.torrent by 2.094316 seconds

Feb 15, 2019 03:58:23 PM | Debug | 
WebClient(HttpWebClient).GetBytes(Url:https://rarbg.to/download.php?id=5gs7zj9&f=jackett.torrent)

Feb 15, 2019 03:58:23 PM | Debug | 
WebClient(HttpWebClient): Returning Redirect => https://rarbg.to/torrent/5gs7zj9

Feb 15, 2019 03:58:23 PM | Debug | 
WebClient(HttpWebClient).GetString(Url:https://torrentapi.org/redirect_to_info.php?token=z0y8jkxuhe&p=1_7_6_3_4_9_3__e786b989bf)

Feb 15, 2019 03:58:23 PM | Info | 
Executing action method Jackett.Server.Controllers.BlackholeController.Blackhole (JackettConsole) with arguments (rarbg, Q2ZESjhCUkRxVlV5V1FOQXVkQ1ZTUkJqejRZdk5talZEeWNxMUVQam9mdUpFT2RSQkx5SFRYOTVtTTl5M0h4UXhPa0FpU2RNNlBCeXJBc01TQ294ekFORUY4ZVB3VkZnUUFCRllzQTZ3cGkzUUZWb3Y4MWthd3pSMUZGd0g1S2xvNmhicTgxeWxhaHBIZWxWSXNQWXM5NWd1ZjNUMUl0U2hyeFB0bmVxcDAtZUxVOHIzTncwTHREZmJuZm4zZVRpNWtGOHdIZmQxLVh5WF9rOGpkRTRIYThIVU9RSXk0Y1d3Wkl6WUJDTHg3WGhlbmth, dr1aeadqq7zvkz44kxpx8jzrd963e4zd, American.Dad.S14E14.1080p.WEBRip.x264-TBS[rartv]) - Validation state: Valid


```


Jackett is definitely getting bytes from the protection webpage.
Shouldn't it be possible to wait for the protection to finish to get the actual real torrent content wanted ?

**Comment by kaso17** [comment]
fixed via ad77068a7b5ece2291c125fb7ed756757618878d (default to magnet now)
